intro-HPC/ch_VNC.tex

```diff
@@ -0,0 +1,132 @@
+\chapter{Graphical applications with VNC}
+\label{ch:vnc}
+
+It's important to remember that VNC sessions are permanent. They survive network
+problems and (unintended) connection loss. This means you can logout and go home
+without a problem (like the terminal equivalent \lstinline|screen| or \lstinline|tmux|).
+This also means you don't have to start \lstinline|vncserver| each time you want to connect.
+
+\section{Start the VNC server}
+\label{sec:start-vnc}
+
+First login on the login node (see \autoref{sec:first-time-connection-to-the-hpc})
+or to the worker node where your job has started. % TODO: where is this described?
+
+Now start \lstinline|vcnserver| with:
+
+\begin{prompt}
+%\shellcmd{vncserver -geometry 1920x1080}%
+You will require a password to access your desktops.
+
+Password:%\emph{<{}enter a secure password>{}}%
+Verify:%\emph{<{}enter the same password>{}}%
+Would you like to enter a view-only password (y/n)? %\emph{n}%
+A view-only password is not used
+
+New '%\strong{gligar02}%.gligar.os:%\strong{6}% (%\userid{}%)' desktop is gligar02.gligar.os:6
+
+Creating default startup script /user/home/gent/vsc000/%\userid{}%/.vnc/xstartup
+Creating default config /user/home/gent/vsc000/%\userid{}%/.vnc/config
+Starting applications specified in /user/home/gent/vsc000/%\userid{}%/.vnc/xstartup
+Log file is /user/home/gent/vsc000/%\userid{}%/.vnc/gligar02.gligar.os:6.log
+
+\end{prompt}
+
+When prompted for a password, make sure to enter a secure password: if someone
+can guess your password, they will be able to do anything with your account you can.
+
+Note down the details in bold: the hostname (in the example \lstinline|gligar02|)
+and the port number (in the example \lstinline|6|).
+
+\section{Connecting to the VNC server}
+
+\subsection{VNC on the login nodes}
+
+The VNC server runs on a \strong{specific login node} (in this example \lstinline|gligar02|).
+Make sure you connect to this login node: the domain should be like \lstinline|gligar02.ugent.be|,
+but the number can be different. If you're not sure how to do this, please follow the steps
```

Comment:
Not only the number, if you want to avoid making this Gent-specific.

Maybe make it more general by using `example.vscsite.be` and explain that they need to use the hostname mentioned in the output of `vncserver`?

Reply:
How should users know how to translate `gligar02.gligar.os` (output ov `vncserver` to `gligar02.ugent.be` (domain they need to connect to)? How does this work at other sites?

Reply:
you make a script to start the server (or handle exsting server), and it prints the hostname to connect to

Reply:
Yeah, good luck with getting that in place everywhere...

@redfast00 I'd go with: use the public hostname of the login node where the VNC server was started. If you're not sure how to determine the public hostname, contact \hpcinfo.

For UGent, you can mention: use the output of `hostname -s` + `.ugent.be`, for example `gligar02.ugent.be`

Reply:
@kehoste what are the odds that vnc works on all sites?
make a script, we can install it in gent to start. i can make rpms (eg vsc-cue-doc-tools), and we can make it part of the release of the docs?
@klust @mselwa @gjbex @smoors what are the odds of getting a single rpm with tools to ease the documentation installed on all sites?

Reply:
I found a way to fix this: `ssh -L 5906:gligar01:5906 vsc400000@gligar02.ugent.be` works for connecting to a VNC server on gligar01, so `@login.hpc.ugent.be` will also work.

Reply:
And now users don't have to connect to a specific host

Reply:
you can now ssh to short hostnames from gent loginnodes